File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.cpp
#include <string>
#include <vector>
#include <bitset>
using namespace std;


int Find(int num)
{
	bitset<20> bits(num);
	int count = 0;
	auto str = bits.to_string();

	for (auto s : str)
	{
		if (s == '1') count++;
	}

	return count;
}

int solution(int n) {
    int answer = 0;
    	int start = Find(n);

	while (true)
	{
		n = n + 1;
		if (Find(n) == start)
		{
			answer = n;
			break;
		}
	}
    return answer;
}

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.cs
using System;
using System.Linq;
class Solution 
{
    public int solution(int n) 
   {
        int answer = 0;
        int cnt = Convert.ToString(n, 2).Count(c => c == '1');
        while(true)
        {
            n += 1;
            int find = Convert.ToString(n, 2).Count(c => c == '1');
            if (cnt == find) break;
        }
        return n;
    }
}